### PR TITLE
Assignment 7

### PR DESCRIPTION
**Pull request by smriti-porwal**
Added Routing.

### CODE REVIEW

File: Assignments/src/Components/ErrorPageComponent/error-page.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useRouteError } from "react-router-dom";
+const ErrorComponent = () => {
+  const error = useRouteError();
+  return (
+    <>
+      <h1>OOps! Something wrong happened! </h1>
+      <h1>Error : {error.statusText}</h1>
```
Comment (nitishnivedan) [inline Assignments/src/Components/ErrorPageComponent/error-page.js:7]: for some cases there is no statusText. have an OR condition to access another object.

File: Assignments/src/Services/getTeamDataService.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    teamMembersList.map(async (teamMember) => {
+      const userInformation = await fetch(
+        `https://api.github.com/users/${teamMember}`
+      );
+      return await userInformation.json();
+    })
+  );
+
+export const getTeamMemberData = async (id) => {
+  return (await fetch(`https://api.github.com/users/${id}`)).json();
```
Comment (nitishnivedan) [inline Assignments/src/Services/getTeamDataService.js:19]: should not have this getTeamMemberData in this component. should have a separate file.

can make use of getTeamsData function. and pass the id there as a parameter to that function

File: Assignments/src/Utility/filter.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const searchFilter = (data, searchText) => {
```
Comment (nitishnivedan) [inline Assignments/src/Utility/filter.js:1]: dont need curly braces here. can return directly

File: Assignments/src/Components/TeamMemberComponent/team-member-component.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  const { id } = useParams();
+
+  const [memberData, setMemberData] = useState([]);
+
+  useEffect(() => {
+    getMemberData(id);
+  }, []);
+  async function getMemberData(id) {
+    let member = await getTeamMemberData(id);
+    console.log(member);
```
Comment (shaksham08) [inline Assignments/src/Components/TeamMemberComponent/team-member-component.js:16]: If not used you can remove the console log

File: Assignments/src/Components/NoDataComponent/no-data-component.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const NoDataComponent = ()=>{
+    return <h1>No Data found</h1>
```
Comment (shaksham08) [inline Assignments/src/Components/NoDataComponent/no-data-component.js:2]: you don't need use a return keyword here

**Review by shaksham08 — APPROVED** [review]
Looks good to me just added some comments

**Pull request merged**